tests/temp_adhoc_terms_testt.R

```diff
@@ -0,0 +1,460 @@
+  return(ergm::summary_formula(net_tmp ~ gwesp(0.5,fixed = T)))
+})
+names(tmp) <- unique(net %v% "var_1")
+
+# ERNM Calcs
+ernm::calculateStatistics(net ~ gwesp(alpha = 0.5,homogenous = T,variableName = "var_1"))
+
+# Check that regular ESP is still working:
+ergm::summary_formula(net ~ gwesp(0.5,fixed = T))
+ernm::calculateStatistics(net ~ gwesp(0.5))
+
+# Check the dyad update works :
+tmp_model <- ernm(net ~ gwesp(alpha = 0.5,homogenous = T,variableName = "var_1"),
+                  maxIter = 2,
+                  mcmcSampleSize = 1000,
+                  mcmcBurnIn = 100)
+
+for(i in 1:(net%n%"n")){
+  for(j in 1:(net%n%"n")){
+    if(i==j){next}
+    net_2 <- net
+    net_2[i,j] <- 1 - net_2[i,j] 
+    change_1 <- ernm::calculateStatistics(net_2 ~ gwesp(alpha = 0.5,homogenous = T,variableName = "var_1")) - 
+      ernm::calculateStatistics(net ~ gwesp(alpha = 0.5,homogenous = T,variableName = "var_1"))
+  
+    model <- tmp_model
+    model <- model$m$sample$getModel()
+    model$setNetwork(as.BinaryNet(net))
+    model$calculate()
+    
+    stat1 <- model$statistics()
+    model$dyadUpdate(i,j)
+    stat2 <- model$statistics()
+    change_2 <- stat2-stat1
+    
+    if(abs(change_1 -  change_2) > 10**-3){
+      print("PROBLEM")
+      print(i)
+      print(j)
+      print(change_1)
+      print(change_2)
+      Sys.sleep(1)
+    }
+  }
+}
+
+# Check reuglar gwesp is still working
+tmp_model <- ernm(net ~ nodeMatch("var_3"))
+tmp_model <- ernm(net ~ gwesp(alpha = 0.5,homogenous = T,variableName = "var_2"),
+                  maxIter = 2,
+                  mcmcSampleSize = 1000,
+                  mcmcBurnIn = 100)
+
+m <- tmp_model$m$sampler$getModel()
+for(i in 1:1000){
+  print(i)
+  tmp <- as.BinaryNet(net)
+}
+# after enough calls something in c++ breaks.. 
+
+
+delete.vertex.attribute(net,"na")
+for(i in which(net %v% "var_2"== "2")){
+  for(j in which(net %v% "var_2"== "2")){
+    if(i==j){next}
+    print(i)
+    print(j)
+    net_2 <- net
+    net_2[i,j] <- 1 - net_2[i,j] 
+    change_1 <- tryCatch({ernm::calculateStatistics(net_2 ~ gwesp(alpha = 0.5,homogenous = T,variableName = "var_2")) - 
+      ernm::calculateStatistics(net ~ gwesp(alpha = 0.5,homogenous = T,variableName = "var_2"))},
+      error = function(e){return(10**10)})
+    
+    model <- tmp_model$m$sampler$getModel()
+    
+    model$setNetwork(as.BinaryNet(net))
+    model$calculate()
+    stat1 <- tryCatch({model$statistics()},error = function(e){10**5})
+    model$dyadUpdate(i,j)
+    stat2 <- tryCatch({model$statistics()},error = function(e){10**7})
+    change_2 <- stat2-stat1
+    if(abs(change_1 -  change_2) > 10**-3){
+      print("PROBLEM")
+      print(i)
+      print(j)
+      print(change_1)
+      print(change_2)
+      Sys.sleep(1)
+    }
+  }
+}
+
+# =====================================================================
+# hamming Function
+# =====================================================================
+
+edges = as.edgelist(net)*1
```

Comment:
These tests scenarios are good; however, we should throw an error (or use testthat) when the output is not as expected. For example, check that hamming_calc and calculateStatistics come up with the same values

Reply:
agree - I noticed there was a rejigging of the testing structure in LOLOG - should we go down that route too? I put these here mainly so you could see I did at least SOME testing, not as a long term testing workflow for additional statistics

Reply:
yes, I'd recommend the lolog test structure. It is (I believe) the standard way to structure testing with testthat.

Reply:
did you want to leave that for another pull?

Reply:
yes please - if thats okay - i'll do it next then add more terms in the new testing framework